### DISCUSSION TITLE
Better way to import/export anonymous subscriptions & multireddits

### DISCUSSION THREAD

**Discussion by sjain882** [Docile-Alligator/Infinity-For-Reddit#1619] [Ideas]
### Checklist

- [X] I have used the search function for [open](https://github.com/Docile-Alligator/Infinity-For-Reddit/discussions/categories/ideas) **and** [closed](https://github.com/Docile-Alligator/Infinity-For-Reddit/discussions/categories/completed-ideas) ideas to see if someone else has already submitted the same request.
- [X] I will describe the feature with as much detail as possible.

### Description

Hi,

To address a few flaws with the current setup:

- Importing/Exporting anonymous subscriptions & multireddits still requires you to have set these up with the app's GUI in the first place - would take ages for even medium amount of multireddits. The backups can't be easily modified externally; even if they could it's not great UX.
- You can't import subscriptions & multireddits into Anonymous mode from an account.

It would be ideal to be able to import/export a plaintext list of subreddits & multireddits in settings - would make it much easier to share & back these up, independently of having a reddit account, which is the direction reddit has gone in unless you are using the official app.

Thanks!

### Additional context

Use-case: UX improvement, better backups, no requirement of having a reddit account to sync subs.

### How important is this feature?

Nice to have

**Comment by kpupp** [comment]
This would be great!

**Comment by KyattPL** [comment]
Exactly, plain txt files import / export would be great. Not willing to use the app without it unfortunately, looking at the dates it seems like it won't be solved :<

**Reply by kpupp** [reply]
In the meantime, I just exported everything as a backup, opened up the zip, and had Anthropic (or your LLM of choice) extract each subreddit name from the JSON file. I think I had to teach it what a name/URL looked like, but it wasn't that bad. 

Still would love something simpler than all that.

**Reply by N1N74** [reply]
@kpupp you can use regex search or even a simple search in any good text editors like for eg. sublime text and select all the names of subreddits automatically. Use `ctrl+right/left` to move a word and `ctrl+shift+left/right` to select that word, then simply copy the selection and paste in another document to get list.
Best not to rely on AI for such tasks imho. I've tried this and for infinity it works using a simple search itself. I haven't been able to get the reverse working, i.e. import into infinity, but I have imported this way into stealth before.

**Comment by N1N74** [comment]
+1

Stealth reddit does this already and I think Infinity should also. I've imported at least 1000 subreddits in stealth but haven't been able to do so on infinity, UI for subscription of a simple list of subreddits can be extremely useful and appreciated.

Alternatively, we can also try making imports from stealth possible which is just a json of subreddits, though I understand why that maynot be preferable to develop.

**Comment by N1N74** [comment]
For backups, instead of password protected archive that stores json, if we had a single json file, it'd be easier to import/export. We can just add new entries to the json file, like how it works for stealth. For compressed files, if I change anything inside the archive, the import process crashes and I haven't figured out how to do this without causing that. Would skip the unnecessary troubleshooting and guesswork if we had that instead. Fault tolerance to empty id fields in the json will also be very good.
